ScraperFC/SofaScore.py

```diff
@@ -1,144 +1,454 @@
+        fig.set_facecolor('white')
+
+        ax.bar(match_momentum_df.minute, match_momentum_df.value, color=plot_colors)
+        ax.axvline(45.5, ls=':')
+        ax.set_xlabel('Minutes')
+        ax.set_xticks(range(0,91,10))
+        ax.set_xlim(0,91)
+
+        plt.gca()
+        ax.spines[['top', 'right', 'left']].set_visible(False)
+        ax.set_yticks([])
+        plt.savefig(f'{match_id}_match_momentum.png', bbox_inches='tight')
+        
+        return fig, ax
+    
+    def get_match_id(self, match_url):
+        """Get match id for any match
+
+        Args:
+            match_url (string): Full link to a SofaScore match
+
+        Returns:
+            string: Match id for a SofaScore match. Used in Urls
+        """
+        match_id = match_url.split('#')[-1]
+        return match_id
+
+    def scrape_league_stats(self, tournament, season, accumulation='total', selected_positions=['Goalkeepers', 'Defenders', 'Midfielders', 'Forwards']):
+        """Get every player statistic that can be asked in league pages on SofaScore.
+
+        Args:
+            tournament (string): Name of the competition
+            season (string): Season selected
+            accumulation (str, optional): Value of the filter accumulation. Can be "per90" and "perMatch". Defaults to 'total'.
+            selected_positions (list, optional): Value of the filter positions. Defaults to ['Goalkeepers', 'Defenders', 'Midfielders', 'Forwards'].
+
+        Returns:
+            DataFrame: DataFrame with each row corresponding to a player and the columns are the fields defined on get_league_stats_fields()
+        """
+        df = pd.DataFrame()
+        
+        offset = 0
+
+        headers = self.get_requests_headers()
+        positions = self.get_positions(selected_positions)
+        season_fields = self.get_league_stats_fields()
+        fields = self.concatenate_fields(season_fields)
+        positions = self.get_positions(selected_positions)
+        season = self.get_seasons(season, tournament)
+        tournaments_ids = self.get_tournaments_ids()
+
+        for i in range(0,100):
+            url = f'https://api.sofascore.com/api/v1/unique-tournament/{tournaments_ids[tournament]}/season/{season}/statistics?limit=100&order=-rating&offset={offset}&accumulation={accumulation}&fields={fields}&filters=position.in.{positions}'
+            response = requests.get(url, headers=headers)
+            new_df = pd.DataFrame(response.json()['results'])
+            new_df['player'] = new_df.player.apply(pd.Series)['name']
+            new_df['team'] = new_df.team.apply(pd.Series)['name']
+            df = pd.concat([df, new_df])
+            
+            page = response.json().get('page')
+            pages = response.json().get('pages')
+
+            if page == pages:
+                print('End of the pages')
+                break
+            offset += 100
+        
+        return df
+
+    def match_momentum(self, match_url):
```

Comment:
This should just return `match_momentum_df`. ScraperFC is meant to return data in the rawest form possible so it can be used for any sort of analysis.